extensions/amp-story-shopping/0.1/test/test-amp-story-shopping-tag.js

```diff
@@ -38,5 +43,27 @@ describes.realWin(
       expect(() => shoppingTag.layoutCallback()).to.not.throw();
       expect(shoppingTag.isLayoutSupported(Layout.CONTAINER)).to.be.true;
     });
+
+    it('should process config data and set text container content if data not null', () => {
+      shoppingTag.element.setAttribute('data-tag-id', 'sunglasses');
+      const shoppingState = {
+        'sunglasses': {'product-title': 'Spectacular Spectacles'},
+      };
+      shoppingTag.updateShoppingTag_(shoppingState);
+      expect(shoppingTag.element.textContent).to.equal(
+        'Spectacular Spectacles'
+      );
+    });
+
+    it('should not process config data and set text container content if id not found is null', () => {
+      shoppingTag.element.setAttribute('data-tag-id', 'hat');
+      const shoppingState = {
+        'sunglasses': {'product-title': 'Spectacular Spectacles'},
+      };
+      shoppingTag.updateShoppingTag_(shoppingState);
+      expect(shoppingTag.element.textContent).to.not.equal(
+        'Spectacular Spectacles'
+      );
```

Comment:
Is `shoppingTag.element.textContent` expected to be defined at all in this case? If it's expected to be undefined/null, then maybe you can instead do something like:
`expect(shoppingTag.element.textContent).to.be.undefined;`

Reply:
Good point, but it actually gives an empty string, I have updated the test accordingly:
`expect(shoppingTag.element.textContent).to.be.empty;`

[Further reading on why textContent returns empty string instead of null](https://developer.mozilla.org/en-US/docs/Web/API/Node/textContent)